app/services/notify_service.rb

```diff
@@ -216,7 +214,7 @@ def call(recipient, type, activity)
     return if drop?
 
     @notification.filtered = filter?
-    @notification.group_key = notification_group_key
+    @notification.set_group_key!
```

Comment:
Is this the only place in the codebase that a notification is created? If so, and/or if we always want to have a key set, a further refactor here would be to make this a callback on the model and delete this line from the service.

Separately - if we're at a point where the notifications stuff is more or less settled down feature wise, it might be good to do a pass over the notification model (which has gotten large) and look for opportunities to pull out classes either related to key calculation, or grouping in general, the specialized pagination, etc.

Reply:
This might make sense, but as part of a future refactor PR for notifications